packages/space-opera/src/components/model_viewer_snippet/components/validation_utils.ts

```diff
@@ -0,0 +1,211 @@
+/**
+ * @license
+ * Copyright 2021 Google LLC. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the 'License');
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an 'AS IS' BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+// Code extracted from:
+// https://github.com/donmccurdy/three-gltf-viewer/
+
+import {validateBytes} from 'gltf-validator';
+import {LoaderUtils, LoadingManager, WebGLRenderer} from 'three';
+import {DRACOLoader} from 'three/examples/jsm/loaders/DRACOLoader.js';
+import {GLTFLoader} from 'three/examples/jsm/loaders/GLTFLoader.js';
+import {KTX2Loader} from 'three/examples/jsm/loaders/KTX2Loader.js';
+
+const SEVERITY_MAP = ['Errors', 'Warnings', 'Infos', 'Hints'];
+
+/**
+ * Loads a gltf provided by the user.
+ * Passes the gltf to be validated.
+ */
+export async function validateGltf(url: string) {
+  const gltf = await loadGltf(url);
+  return await validate(url, gltf);
+}
+
+/**
+ * Loads the gltf using the GLTF Loader.
+ *
+ * @param url The gltf's url in state.entities.gltf.gltfUrl, which is set
+ *     when the model is initially loaded.
+ */
+async function loadGltf(url: string) {
```

Comment:
Is this actually necessary? It seems odd since we already have this code inside model-viewer. If I'm reading this correctly, it seems like all this gltf does is get passed down through functions until it displays the file's metadata, which doesn't seem super necessary. Can we just cut a bunch of code out by only validating?

Reply:
If we think this is important, it would probably be better to add a `getMetadata` method on model-viewer and call that.

Reply:
👍